### DISCUSSION TITLE
Is there a way to get address from contract instance?

### DISCUSSION THREAD

**Discussion by b-pmcg** [wevm/viem#535] [Question] [answered]
Before wagmi v1 we were using the wagmi cli to generate actions such as a `getMyContract` function. Then I could get the address from the contract instance like `const addr = getMyContract().address`. Now that wagmi has migrated to viem, the cli-generated getContract functions return a [viem contract instance](https://viem.sh/docs/contract/getContract.html) which does not have the `address` property. 

Is there any other way to retrieve the contract address with the contract instance or something similar?

**Comment by tmm** [comment, marked as answer]
Would accept a contribution for this! Should be very straightforward to add.

**Comment by Zimtente** [comment]
any updates on that?

**Comment by milaabl** [comment]
@Zimtente, looks like a PR for that has already been merged: https://github.com/wagmi-dev/viem/pull/619

You should be able to access both `.address` and `.abi` on the `Contract` instance
